kernel_gateway/tests/test_gatewayapp.py

```diff
@@ -56,7 +71,69 @@ def test_headless(self):
         self.assertEqual(response.code, 404)
 
     @gen_test
-    def test_kernel(self):
+    def test_auth_token(self):
+        '''All kernel endpoints should check the configured auth token.'''
+        # Set token requirement
+        app = self.get_app()
+        app.settings['kg_auth_token'] = 'fake-token'
+
+        # Request a kernel without the token
+        response = yield self.http_client.fetch(
+            self.get_url('/api/kernels'),
+            method='POST',
+            body='{}',
+            raise_error=False
+        )
+        self.assertEqual(response.code, 401)
+
+        # Request with the token now
+        response = yield self.http_client.fetch(
+            self.get_url('/api/kernels'),
+            method='POST',
+            body='{}',
+            headers={'Authorization': 'Token fake-token'},
+            raise_error=False
+        )
+        self.assertEqual(response.code, 201)
+
+        kernel = json_decode(response.body)
+        # Request kernel info without the token
+        response = yield self.http_client.fetch(
+            self.get_url('/api/kernels/'+url_escape(kernel['id'])),
+            method='GET',
+            raise_error=False
+        )
+        self.assertEqual(response.code, 401)
+
+        # Now with it
+        response = yield self.http_client.fetch(
+            self.get_url('/api/kernels/'+url_escape(kernel['id'])),
+            method='GET',
+            headers={'Authorization': 'Token fake-token'},
+            raise_error=False
+        )
+        self.assertEqual(response.code, 200)
+
+        # Request websocket connection without the token
+        ws_url = 'ws://localhost:{}/api/kernels/{}/channels'.format(
+            self.get_http_port(),
+            url_escape(kernel['id'])
+        )
+        # No option to ignore errors so try/except
+        try:
+            ws = yield websocket_connect(ws_url)
+        except Exception as ex:
+            self.assertEqual(ex.code, 401)
```

Comment:
Should this test to make sure that the `Exception` does get raised? (it won't test it if no `Exception` occurred)